pretyping/pretyping.ml

```diff
@@ -629,8 +629,16 @@ let rec pretype k0 resolve_tc (tycon : type_constraint) (env : ExtraEnv.t) evdre
         match tycon with
         | Some ty -> ty
         | None ->
-          new_type_evar env evdref loc in
-        { uj_val = e_new_evar env evdref ~src:(loc,k) ~naming ty; uj_type = ty }
+           new_type_evar env evdref loc in
+      let store =
+        if Flags.is_program_mode () then
```

Comment:
It would be great if we could avoid these kind of hidden switches in a low-level component such as the pretyper.

Reply:
I agree but there is a gazillion of `Flags.is_program_mode` already in there...

Reply:
It moved up from unification to here with this patch, so already a bit higher-level. I think the next step would be to actually change the switch to a test of an "evars_to_goals"  boolean or somesuch, which would tag user-written "_" as evars standing for potential subgoals (for refine) or obligations (for Program).